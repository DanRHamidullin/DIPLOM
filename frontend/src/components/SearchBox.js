import React, { useState } from 'react'
import { Button, Form } from 'react-bootstrap'
import { useHistory } from 'react-router-dom'

function SearchBox() {

    //useState - это хук, который позволяет добавлять состояние React к компонентам-функциям.
    //Обьявили переменную состояния keyword и функцию обновления значения setKeyword, и инициализировали его начальное состояние через useState('').
    //Этот хук возвращает нам текущее состояне keyword
    const [keyword, setKeyword] = useState('')

    //useHistory - это хук React Router, который предоставляет доступ к экземпляру истории, который используется для навигации.
    //То есть дальше мы можем использовать этот обьект для перехода на другую страницу через history.push()
    let history = useHistory()

    const submitHandler = (e) => {
        e.preventDefault()
        if (keyword) { //Здесь переходим по введеной строке поиска.
            history.push(`/?keyword=${keyword}&page=1`)
        } else { //Здесь остаемся на месте, так как поле поиска пустое.
            history.push(history.push(history.location.pathname))
        }
    }
    return (
        <Form onSubmit={submitHandler} inline>
            <Form.Control
                type='text'
                name='q'
                onChange={(e) => setKeyword(e.target.value)}
                className='mr-sm-2 ml-sm-5'
            ></Form.Control>

            <Button
                type='submit'
                variant='outline-success'
                className='p-2'
            >
                Поиск
            </Button>
        </Form>
    )
}

export default SearchBox
