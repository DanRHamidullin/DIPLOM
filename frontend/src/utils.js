export function formatUserName(username) {
 return 'login: ' + username;
}