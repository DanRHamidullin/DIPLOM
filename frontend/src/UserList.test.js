import { formatUserName } from "./utils";
import { render, screen, waitFor } from '@testing-library/react';
import UserList from "./UserList";
import '@testing-library/jest-dom'
import renderer from "react-test-renderer";
import React from "react";
import SearchBox from "./components/SearchBox";


//Блок describe - это набор тестов, а test - это тестовый кейс. В наборе тестов может быть несколько тестовых кейсов.
//Внутри тестового примера мы пишем утверждения (например, expect в Jest), которые проверяют успешность (зеленый)
// или ошибочность (красный) утверждения.

//Это модульный тест, который проверяет, есть преборазуюется ли username строка в вид: login: username.
describe('utils test', () => {
 test('formatUserName adds "login" at the beginning of the username', () => {
   expect(formatUserName('jc')).toBe('login: jc');
 });
});

//Мы используем функцию рендеринга mount и визуализируем отдельный компонент изолированно.
//Этот тест завершится неудачно только в том случае, если есть ошибка компиляции или ошибка
// в функциональном компоненте, которая препятствует его визуализации. Здесь не утверждений.
describe('UserList component', () => {
 test('it renders', () => {
   render(<UserList />);
 });
})

//Тот же тест, что и выше, но с утверждением.
//Проверка что компонент был срендерен, но также поиск элемента, присутствующего в DOM, с текстом «Users:»
describe('UserList component 1', () => {
 test('it renders 1', () => {
   render(<UserList />);

   expect(screen.getByText('Users:')).toBeInTheDocument();
 });
})

//Здесь происходит асинхронная операция (fetch), поэтому мы дожидаемся, когда завершится вызов у await и уже смотрим утверждение.
//Если не использовать await, то тест не выполнится, потому что вызов ассинхронной операции еще не выполнен.
describe('UserList component 2', () => {
 test('it displays a list of users', async () => {
   render(<UserList />);


   const userList = await waitFor(() => screen.getByTestId('user-list'));
   expect(userList).toBeInTheDocument();
 });
});


//Тест на основе снимка.
describe('UserList component render', () => {
    test("render correctly UserList component", () => {
        const UserListComponent = renderer.create(<UserList/>);
        let tree = UserListComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });
})