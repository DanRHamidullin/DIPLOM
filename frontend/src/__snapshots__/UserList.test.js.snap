// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList component render render correctly UserList component 1`] = `
<div
  className="App"
>
  <div>
    Users:
  </div>
  <div>
    Loading users...
  </div>
</div>
`;
